src/molinillo_solver.cr

```diff
@@ -162,11 +172,11 @@ module Shards
     end
 
     def dependencies_for(specification : S) : Array(R)
-      return specification.dependencies if specification.name == "crystal"
-      return specification.dependencies if @ignore_crystal_version
+      return apply_overrides(specification.dependencies) if specification.name == "crystal"
```

Comment:
I'd extract `dependencies = apply_overrides(specification.dependencies)` to avoid repeating the same call